Source/DafnyCore/Verifier/Translator.cs

```diff
@@ -1274,14 +1274,15 @@ internal class PredefinedDecls {
       FuelContext oldFuelContext = this.fuelContext;
       this.fuelContext = FuelSetting.NewFuelContext(dd);
 
-      if (dd.Var != null) {
-        AddWellformednessCheck(dd);
-        currentModule = dd.EnclosingModuleDefinition;
-        // Add $Is and $IsAlloc axioms for the newtype
-        AddRedirectingTypeDeclAxioms(false, dd, dd.FullName);
-        AddRedirectingTypeDeclAxioms(true, dd, dd.FullName);
-        currentModule = null;
-      }
+      dd.Var ??= new BoundVar(dd.tok, "x", dd.BaseType) { IsTypeExplicit = true };
+      dd.Constraint ??= Expression.CreateBoolLiteral(dd.tok, true);
+
+      AddWellformednessCheck(dd);
+      currentModule = dd.EnclosingModuleDefinition;
+      // Add $Is and $IsAlloc axioms for the newtype
+      AddRedirectingTypeDeclAxioms(false, dd, dd.FullName);
+      AddRedirectingTypeDeclAxioms(true, dd, dd.FullName);
+      currentModule = null;
```

Comment:
Well identified! These type axioms are what's missing. However, it's not a good idea to change the AST at this time. Also, I think it's fine to continue to omit a well-formedness check for these var-less newtypes (hence, addressing your question in the PR comment).

So, I suggest making the `if (dd.Var != null)` control only whether or not `AddWellformednessCheck` is called. Then, do the other lines in either case (as you have in your new L1281-L1285).

Then, adjust `AddRedirectingTypeDeclAxioms` so that it's able to handle the `dd.Var == null` case. Instead of `dd.Var.Type` (which occurs many times in that method), use `baseType` defined as

``` c#
var baseType = dd.Var != null ? dd.Var.Type : ((NewtypeDecl)(object)dd).BaseType;
```

One of the method's preconditions thus needs to change to

``` c#
Contract.Requires((dd.Var != null && dd.Constraint != null) || dd is NewtypeDecl);
```

There are two other occurrences of `dd.Var` that you'll need to handle in some way.

FInally, if `dd.Constraint` is `null`, use `Bpl.Expr.True` instead of `etran.TrExpr(dd.Constraint)`.